tools/harness_capability_profiles.py

```
@@ -606,9 +653,116 @@ def markdown_table_rows(markdown: str, heading: str) -> list[dict[str, str]]:
+                TABLE_PARSE_ERROR_DETAIL: f"expected {len(headers)} cells, got {len(cells)}",
+            }
+        )
     return rows
 
 
+def validate_required_markdown_sections(markdown: str, required_sections: list[str], prefix: str, path: Path) -> list[CheckResult]:
+    results: list[CheckResult] = []
+    headings = {line.strip().casefold() for line in markdown.splitlines()}
+    for section in required_sections:
+        if section.casefold() not in headings:
+            results.append(CheckResult(f"{prefix}:section:{section.removeprefix('## ').casefold().replace(' ', '_')}", False, f"missing section: {section}", path.as_posix()))
```

Ruff (0.15.12)

[warning] Use a list comprehension to create a transformed list

(PERF401)